xwiki-platform-core/xwiki-platform-rest/xwiki-platform-rest-server/src/main/java/org/xwiki/rest/internal/resources/BaseSearchResult.java

```diff
@@ -137,6 +155,63 @@ protected List<SearchResult> search(List<SearchScope> searchScopes, String keywo
         }
     }
 
+    private List<SearchResult> searchPagesSolr(List<SearchScope> searchScopes, String keywords, String wikiName,
+        String space, int number, int start, String orderField, String order, Boolean withPrettyNames,
+        Boolean isLocaleAware) throws QueryException, XWikiException
+    {
+        if (StringUtils.isBlank(keywords)) {
+            return List.of();
+        }
+
+        List<String> filterQueries = new ArrayList<>();
+        filterQueries.add("type:DOCUMENT");
+        filterQueries.add("hidden:false");
+        if (StringUtils.isNotBlank(wikiName)) {
+            filterQueries.add("wiki:" + this.solrUtils.toCompleteFilterQueryString(wikiName));
+        }
+        if (StringUtils.isNotBlank(space)) {
+            filterQueries.add("space_exact:" + this.solrUtils.toCompleteFilterQueryString(space));
+        }
+
+        String escapedKeyWords = this.solrUtils.toFilterQueryString(keywords);
+        Set<SearchScope> supportedScopes = EnumSet.of(TITLE, NAME, CONTENT);
+        String queryString = searchScopes.stream()
+            .filter(supportedScopes::contains)
+            .map(scope -> switch (scope) {
+                // Consider matches in the title as way more important.
+                case TITLE -> "(title:" + escapedKeyWords + "*)^4";
+                // Prefer matching in the name over the spaces, but only if the name is not "WebHome".
+                case NAME -> "spaces:" + escapedKeyWords + "* OR (name:" + escapedKeyWords + "* -name_exact:WebHome)^2";
+                case CONTENT -> "doccontent:" + escapedKeyWords;
+                default -> throw new IllegalStateException("Unexpected value: " + scope);
+            })
+            .collect(Collectors.joining(" OR "));
+
+        if (queryString.isEmpty()) {
+            // No supported scope.
+            return List.of();
+        }
+
+        if (Boolean.TRUE.equals(isLocaleAware)) {
+            // TODO: this might not be the right way to filter by language.
+            Locale currentLocale = this.localizationContext.getCurrentLocale();
+            filterQueries.add("locale:(\"\" OR %s OR %s)".formatted(currentLocale, currentLocale.getLanguage()));
+        }
+
+        Query query = this.queryManager.createQuery(queryString, "solr");
+        query.setLimit(number * 2);
```

Comment:
This deserves some explanation.

Reply:
I copied this from the old database code where the explanation was that we could match both the original and the translation. It's entirely possibly that we actually don't need this here because we match the documents correctly.